hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/HASecurityUtils.java

```diff
@@ -281,4 +300,83 @@ private static void persistSubCACertificate(OzoneConfiguration config,
     certCodec.writeCertificate(certificateHolder);
   }
 
-}
\ No newline at end of file
+  /**
+   * Create Server TLS parameters required for Ratis Server.
+
+   * @return Parameter map set with TLS config.
+   */
+  public static Parameters createSCMServerTlsParameters(
+      GrpcTlsConfig grpcTlsConfig) {
+    Parameters parameters = new Parameters();
+
+    if (grpcTlsConfig != null) {
+      GrpcConfigKeys.Server.setTlsConf(parameters, grpcTlsConfig);
+      GrpcConfigKeys.Admin.setTlsConf(parameters, grpcTlsConfig);
+      GrpcConfigKeys.Client.setTlsConf(parameters, grpcTlsConfig);
+      GrpcConfigKeys.TLS.setConf(parameters, grpcTlsConfig);
+    }
+
+    return parameters;
+  }
+
+  /**
+   * Create GrpcTlsConfig.
+   * @param conf
+   * @param certificateClient
+   * @return
+   */
+  public static GrpcTlsConfig createSCMRatisTLSConfig(SecurityConfig conf,
+      CertificateClient certificateClient) {
+    if (conf.isSecurityEnabled() && conf.isGrpcTlsEnabled()) {
+      return new GrpcTlsConfig(
+          certificateClient.getPrivateKey(), certificateClient.getCertificate(),
+          certificateClient.getCACertificate(), true);
+    }
+    return null;
+  }
+
+  /**
+   * Submit SCM certs request to ratis using RaftClient.
+   * @param raftGroup
+   * @param tlsConfig
+   * @param message
+   * @return SCMRatisResponse.
+   * @throws Exception
+   */
+  public static SCMRatisResponse submitScmCertsToRatis(RaftGroup raftGroup,
+      GrpcTlsConfig tlsConfig, Message message) throws Exception {
+    final RaftProperties properties = new RaftProperties();
+
+    // TODO: GRPC TLS only for now, netty/hadoop RPC TLS support later.
+    RaftConfigKeys.Rpc.setType(properties, RpcType.valueOf("GRPC"));
+
+
+    // For now not making anything configurable, RaftClient  is only used
+    // in SCM for DB updates of sub-ca certs go via Ratis.
+    RaftClient.Builder builder =  RaftClient.newBuilder()
+        .setRaftGroup(raftGroup)
+        .setLeaderId(null)
+        .setProperties(properties)
+        .setRetryPolicy(
+            RetryPolicies.retryUpToMaximumCountWithFixedSleep(120,
+                TimeDuration.valueOf(500, TimeUnit.MILLISECONDS)));
+
+    if (tlsConfig != null) {
+      Parameters parameters = new Parameters();
+      GrpcConfigKeys.Client.setTlsConf(parameters, tlsConfig);
+      builder.setParameters(parameters);
+    }
+
+    RaftClient raftClient =  builder.build();
+
+    CompletableFuture<RaftClientReply> future =
+        raftClient.async().send(message);
+
+    RaftClientReply raftClientReply = future.get();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you are using CompletableFuture.get() without setting its timeout parameter. Be sure to set timeout values or handle timeout exceptions to prevent threads from waiting for a process to end. This can lead to downtime for your customers. [here](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get-long-java.util.concurrent.TimeUnit-).